shap/plots/_bar.py

```diff
@@ -97,16 +110,21 @@ def bar(shap_values, max_display=10, order=Explanation.abs, clustering=None, clu
     else:
         partition_tree = clustering
     if partition_tree is not None:
-        assert partition_tree.shape[1] == 4, "The clustering provided by the Explanation object does not seem to be a partition tree (which is all shap.plots.bar supports)!"
+        if partition_tree.shape[1] != 4:
+            raise TypeError(
+                "The clustering provided by the Explanation object does not seem to be a "
+                "partition tree, which is all shap.plots.bar supports."
+            )
     op_history = cohort_exps[0].op_history
     values = np.array([cohort_exps[i].values for i in range(len(cohort_exps))])
 
     if len(values[0]) == 0:
-        raise Exception("The passed Explanation is empty! (so there is nothing to plot)")
+        raise ValueError("The passed Explanation is empty! (so there is nothing to plot)")
 
-    # we show the data on auto only when there are no transforms
+    # we show the data on auto only when there are no transforms (excluding getitem calls)
     if show_data == "auto":
-        show_data = len(op_history) == 0
+        transforms = [t for t in op_history if t.get("name") != "__getitem__"]
+        show_data = len(transforms) == 0
```

Comment:
Here is one functional change that I needed to make in order to reproduce the plots in the API example notebook.

In the API example notebook, many plots have show_data=True. It looks like this behaviour was then changed at some point to only show data values when there are no transforms applied.

However, I think `show_data` still makes sense as `True` for a single explanation when the only operation applied has been `__getitem__`.

### Before (show_data = False)
<details>

![image](https://github.com/shap/shap/assets/71127464/fe50e9e8-b315-419e-8cab-029bdd9aca54)

</details>

### After (show_data = True)
<details>

![image](https://github.com/shap/shap/assets/71127464/eb216e88-dd01-427d-8b6e-a33957cdb8f0)

</details>